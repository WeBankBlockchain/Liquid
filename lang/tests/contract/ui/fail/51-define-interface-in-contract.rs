use liquid_lang as liquid;

#[liquid::contract]
mod noop1 {
    use super::*;

    #[liquid::interface(name = auto)]
    mod foo {
        extern "liquid" {
            fn foo(&self);
        }
    }

    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self) {}
    }
}

fn main() {}
