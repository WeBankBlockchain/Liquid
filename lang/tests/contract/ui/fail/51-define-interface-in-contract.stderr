error: definition of interface `foo` must be placed before definition of the contract
  --> $DIR/51-define-interface-in-contract.rs:8:5
   |
8  | /     mod foo {
9  | |         extern "liquid" {
10 | |             fn foo(&self);
11 | |         }
12 | |     }
   | |_____^
